vic/drivers/classic/src/parse_output_info.c

```diff
@@ -31,79 +31,126 @@
-    size_t               noutfiles;
+    extern option_struct       options;
+    extern global_param_struct global_param;
+
+    char                       cmdstr[MAXSTRING];
+    char                       optstr[MAXSTRING];
+    char                       flgstr[MAXSTRING];
+    short int                  streamnum;
+    char                       varname[MAXSTRING];
+    int                        outvarnum;
+    char                       format[MAXSTRING];
+    char                       typestr[MAXSTRING];
+    int                        type;
+    char                       multstr[MAXSTRING];
+    double                     mult;
+    size_t                     nstreams;
+    size_t                     nvars;
+    unsigned int               nextagg;
 
     strcpy(format, "*");
 
     /** Read through global control file to find output info **/
 
     fgets(cmdstr, MAXSTRING, gp);
 
-    outfilenum = -1;
+    streamnum = -1;
     outvarnum = 0;
 
     // Count the number of output files listed in the global param file
-    noutfiles = count_n_outfiles(gp);
+    nstreams = count_n_outfiles(gp);
 
     // only parse the output info if there are output files to parse
-    if (noutfiles > 0) {
-        options.Noutfiles = noutfiles;
+    if (nstreams > 0) {
+        options.Noutstreams = nstreams;
 
-        *out_data_files = calloc(options.Noutfiles, sizeof(*(*out_data_files)));
-        if (*out_data_files == NULL) {
+        output_streams = calloc(options.Noutstreams, sizeof(*(output_streams)));
+        if (output_streams == NULL) {
+            log_err("Memory allocation error in parse_output_info().");
+        }
+        out_data_files = calloc(options.Noutstreams, sizeof(*(out_data_files)));
+        if (out_data_files == NULL) {
             log_err("Memory allocation error in parse_output_info().");
         }
-        init_output_list(out_data, false, "%.4g", OUT_TYPE_FLOAT, 1);
-
-        // PRT_SNOW_BAND is ignored if options.Noutfiles > 0
-        options.PRT_SNOW_BAND = false;
 
         while (!feof(gp)) {
             if (cmdstr[0] != '#' && cmdstr[0] != '\n' && cmdstr[0] != '\0') {
                 sscanf(cmdstr, "%s", optstr);
 
                 if (strcasecmp("OUTFILE", optstr) == 0) {
-                    outfilenum++;
-                    if (outfilenum >= (short int)options.Noutfiles) {
+                    streamnum++;
+                    if (streamnum >= (short int) options.Noutstreams) {
                         log_err("Found too many output files, was expecting "
-                                "%zu but found %hu", options.Noutfiles,
-                                outfilenum);
+                                "%zu but found %hu", options.Noutstreams,
+                                streamnum);
                     }
-                    sscanf(cmdstr, "%*s %s",
-                           (*out_data_files)[outfilenum].prefix);
+                    sscanf(cmdstr, "%*s %s", out_data_files[streamnum]->prefix);
```

Comment:
preface `out_data_files` with a `*` here and in all other instances in this function...